### PR TITLE
Prescription Glasses Upgrades and Sec Beret Sprite Fix

### LINKED ISSUE #1389: IPC sprite bug
Security beret has a sprite bug when worn on IPCs 
capture
only when looking north The beret has weird orange lines on the top

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Adds the ability to combine prescription glasses with a couple other glasses
- Mesons + Prescription Glasses = Prescription Mesons
 - You cannot upgrade the Night Vision Mesons with prescription glasses
- Sunglasses + Prescription Glasses  = Prescription Sunglasses
- HUDSunglasses + Prescription Glasses = (NEW!) Prescription HUDSunglasses.

Adds the ability to use a screwdriver to disassemble (separate) the above upgraded glasses.
- Returns the prescription glasses and the mesons/sunglasses/HUDSunglasses

Also fixes the sprite for the north-facing Security Officer Beret randomly having some orange pixels.
- Fixes #1389

No new sprites, as none were needed.

### CODE REVIEW

**Comment by TheDZD** [comment]
:+1:

**Comment by tigercat2000** [comment]
... wouldn't it be easier to just set their prescription to 1 instead of making a whole new subtype?

**Comment by FalseIncarnate** [comment]
There already was prescription subtypes for sunglasses and mesons. I just made one for the HUDSunglasses to maintain consistency with the others, plus you can then make checks for them more specifically.

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -23,6 +23,21 @@
...
+		return
+	var/mob/living/carbon/human/H = user
+	if(istype(O, /obj/item/clothing/glasses/regular))
+		H.unEquip(src)
+		H.unEquip(O)
+		var/obj/item/clothing/glasses/meson/prescription/P = new /obj/item/clothing/glasses/meson/prescription(get_turf(H))
+		H << "You fit \the [src.name] with lenses from \the [O.name]."
+		H.put_in_hands(P)
+		del(O)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: qdel over del.

File: code/modules/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -40,6 +61,24 @@
...
+		H <<"You can't possibly imagine how adding more lenses would improve \the [src.name]."
+		return
+	if(istype(O, /obj/item/weapon/screwdriver))
+		H.unEquip(src)
+		var/obj/item/clothing/glasses/meson/M = new /obj/item/clothing/glasses/meson(get_turf(H))
+		var/obj/item/clothing/glasses/regular/R = new /obj/item/clothing/glasses/regular(get_turf(H))
+		H << "You salvage the prescription lenses from \the [src.name]."
+		H.put_in_hands(M)
+		H.put_in_hands(R)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: qdel

File: code/modules/clothing/glasses/glasses.dm [hunk 3]
```diff
@@ -145,6 +184,21 @@
...
+		return
+	var/mob/living/carbon/human/H = user
+	if(istype(O, /obj/item/clothing/glasses/regular))
+		H.unEquip(src)
+		H.unEquip(O)
+		var/obj/item/clothing/glasses/sunglasses/prescription/P = new /obj/item/clothing/glasses/sunglasses/prescription(get_turf(H))
+		H << "You fit \the [src.name] with lenses from \the [O.name]."
+		H.put_in_hands(P)
+		del(O)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: all the qdel's

File: code/modules/clothing/glasses/glasses.dm [hunk 4]
```diff
@@ -241,6 +301,24 @@
...
+		H <<"You can't possibly imagine how adding more lenses would improve \the [src.name]."
+		return
+	if(istype(O, /obj/item/weapon/screwdriver))
+		H.unEquip(src)
+		var/obj/item/clothing/glasses/sunglasses/S = new /obj/item/clothing/glasses/sunglasses(get_turf(H))
+		var/obj/item/clothing/glasses/regular/R = new /obj/item/clothing/glasses/regular(get_turf(H))
+		H << "You salvage the prescription lenses from \the [src.name]."
+		H.put_in_hands(S)
+		H.put_in_hands(R)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: it's a qdel world.

File: code/modules/clothing/glasses/glasses.dm [hunk 5]
```diff
@@ -265,6 +343,47 @@
...
+		return
+	var/mob/living/carbon/human/H = user
+	if(istype(O, /obj/item/clothing/glasses/regular))
+		H.unEquip(src)
+		H.unEquip(O)
+		var/obj/item/clothing/glasses/sunglasses/sechud/prescription/P = new /obj/item/clothing/glasses/sunglasses/sechud/prescription(get_turf(H))
+		H << "You fit \the [src.name] with lenses from \the [O.name]."
+		H.put_in_hands(P)
+		del(O)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: Return of the Del: qdel's killed them edition.

File: code/modules/clothing/glasses/glasses.dm [hunk 6]
```diff
@@ -265,6 +343,47 @@
...
+		H <<"You can't possibly imagine how adding more lenses would improve \the [src.name]."
+		return
+	if(istype(O, /obj/item/weapon/screwdriver))
+		H.unEquip(src)
+		var/obj/item/clothing/glasses/sunglasses/sechud/S = new /obj/item/clothing/glasses/sunglasses/sechud(get_turf(H))
+		var/obj/item/clothing/glasses/regular/R = new /obj/item/clothing/glasses/regular(get_turf(H))
+		H << "You salvage the prescription lenses from \the [src.name]."
+		H.put_in_hands(S)
+		H.put_in_hands(R)
+		del(src)
```
Comment (Fox-McCloud) [inline code/modules/clothing/glasses/glasses.dm]: cue: qdel horror music.

**Comment by FalseIncarnate** [comment]
Updated to qdel() as advised by Fox

**Comment by Krausus** [comment]
> There already was prescription subtypes for sunglasses and mesons. I just made one for the HUDSunglasses to maintain consistency with the others

It really doesn't seem like consistency with snowflake subtypes is the best way to go about this, especially if it means copy-pasting code across multiple types of glasses. There should only be procs on the base glasses type, with a way to flag types down the line as able to be upgraded. We want *less* snowflake code, not *more.*

**Comment by FalseIncarnate** [comment]
I don't think adding a new spawn-able type exactly qualifies as "snowflake" code, but the definition of what qualifies as that seems to shift a lot.
As for adding prices to the base type, I'd need to then put in checks for every type I want to be changeable and not changeable together. It seemed more readable to do it this way.

**Comment by Krausus** [comment]
> I don't think adding a new spawn-able type exactly qualifies as "snowflake" code

The subtypes weren't snowflakey until you added an upgrade system that uses different copy-pasted procs and creates/destroys objects to, effectively, change **two variables.**

> As for adding prices to the base type, I'd need to then put in checks for every type I want to be changeable and not changeable together.

No, you'd need to add a single variable, set it on upgradable types, and check it in the procs.

> It seemed more readable to do it this way.

Having procs copy-pasted three times *really* seems more readable? And how would you add support for more upgradable eyewear? Copy-paste the procs *some more?*

**Pull request merged**